File: extras/challenges/build-tower-advanced/sol.js
function towerBuilder(nFloors, nBlockSz) {i
  var lastFloor = nFloors * nBlockSz[0] * 2 - nBlockSz[0];
  var tower = [];
  var floor = 0;
  var space = " ";
  var strk = "*";
  for(var i = 0; i < nFloors * nBlockSz[1]; i++) {
    let lenf = "";
    if(i % nBlockSz[1] == 0) floor += 1;
    let width = floor * nBlockSz[0] * 2 - nBlockSz[0];
    let spaceWidth = (lastFloor - width)/2;
    lenf += space.repeat(spaceWidth);
    lenf += strk.repeat(width);
    lenf += space.repeat(spaceWidth);
    console.log(lenf.length, spaceWidth, width, floor);
    tower.push(lenf);
  }

  return tower;
}

const built = towerBuilder(3, [4, 2])

built.forEach((item) => {
  console.log(item);
})

File: extras/challenges/build-tower-advanced/sol.cpp
#include <vector>
#include <string>
#include <iostream>

std::vector<std::string> towerBuilder(int nFloors, std::vector<int> nBlockSz)
{
  int last_floor = nFloors * nBlockSz[0] * 2 - nBlockSz[0];
  std::vector<std::string> tower;
  int floor = 0;
  for(int i = 0; i < nFloors * nBlockSz[1]; i++)
  {
    if(i % nBlockSz[1] == 0) floor += 1;
    int width = floor * nBlockSz[0] * 2 - nBlockSz[0];
    int spaceWidth = (last_floor - width)/2;
    tower.push_back(std::string(spaceWidth, ' ') + std::string(width, '*') + std::string(spaceWidth, ' '));
  }

  return tower;
}

int main()
{
 std::vector<std::string> built = towerBuilder(3, {4,2});
  for(std::string x : built) {
    std::cout << x << std::endl;
  }
}
